### PR TITLE
Update tutorials for dojo v4.0.0 release

### PR DESCRIPTION
**Pull request by agubler**
Work in progress to update the tutorials inline with the v4.0.0 release.

* [x] Update to use `vdom#renderer` over deprecated `ProjectorMixin` to append to the DOM.
* [x] Update initial demos to return `v()` directly from the render function passed to the `renderer`.
* [x] Update routing demo and works to reference new `Outlet` widget rather than HOC and `routes.ts` convention.
* [ ] ~Can we get a stores demo done? Either way think we should remove the injector/container tutorial.~
* [x] Update dependencies after v4.0.0 release (updated to next apart from test-intern)
* [x] Add `@dojo/cli-create-theme` back into dev deps for theme tutorial
* [x] Test all the snippets lines and "other" words (everything!)

With current changes: https://dist-fovybkuhbd.now.sh/

### CODE REVIEW

**Comment by agubler** [comment]
@dylans quite a few words are changing here, would be great to have your review.

File: site/source/tutorials/002_creating_an_application/index.md [hunk 1]
```diff
@@ -28,58 +28,62 @@ You also need to be familiar with TypeScript as Dojo uses it extensively.
...
 * A Dojo application can easily coexist on a page with other content.
 * That content can consist of static assets, a legacy application or even another Dojo application.
 
 {% section %}
 
-## The projector
+## Mounting Your Dojo Application
 
-In [Your first Dojo application](../001_static_content/), we reviewed Dojo's use of a virtual DOM to provide an abstraction between the application and the rendered page. The projector is the component that serves as the intermediary between these two aspects of the application and, as such, has a presence in both the application and the main HTML document. Review these lines in `main.ts`:
+In [Your first Dojo application](../001_static_content/), we reviewed Dojo's use of a virtual DOM to provide an abstraction between the application and the rendered page. To actually render your dojo application, we use the `renderer` function from the `vdom` module. The `renderer` accepts a function that returns virtual DOM (generated either the `w()` or `v()` pragma, more on these later!).
```
Comment (dylans) [inline site/source/tutorials/002_creating_an_application/index.md]: "generated either the" -> "generated by either the"?

File: site/source/tutorials/002_creating_an_application/index.md [hunk 2]
```diff
@@ -28,58 +28,62 @@ You also need to be familiar with TypeScript as Dojo uses it extensively.
...
+{% instruction 'The `mount` operation in `main.ts`:' %}
+
+{% include_codefile 'demo/initial/biz-e-corp/src/main.ts' line:7 %}
 
 {% section %}
 
 ## Widgets
 
-Widgets are the basic building blocks of Dojo's user interface. They combine both the visual and behavioral aspects of a component into a single element. These aspects are encapsulated within the widget's implementation. The widget then exposes properties and methods that allow other components to interact with it. Consider the following diagrams:
+Up until now, we've return virtual DOM that represents HTML elements directly from the `renderer`, however that wouldn't scale very well as an application grow.
```
Comment (dylans) [inline site/source/tutorials/002_creating_an_application/index.md]: Up until now, we've return virtual DOM that represents HTML elements directly from the `renderer`, however that wouldn't scale very well as an application grow. -> Until now, we have returned a virtual DOM that represents HTML elements directly from the `renderer`, however, that would not scale as an application grows.

File: site/source/tutorials/002_creating_an_application/index.md [hunk 3]
```diff
@@ -28,58 +28,62 @@ You also need to be familiar with TypeScript as Dojo uses it extensively.
...
+
+That were widgets come into play, widgets are the basic building blocks of Dojo's user interface. They combine both the visual and behavioral aspects of a component into a single element. These aspects are encapsulated within the widget's implementation. The widget then exposes properties and methods that allow other components to interact with it. Consider the following diagrams:
```
Comment (dylans) [inline site/source/tutorials/002_creating_an_application/index.md]: "That were widgets come into play, widgets are the basic building blocks of Dojo's user interface." -> This is where widgets are beneficial, as widgets are Dojo's basic building blocks for user interfaces"

"The widget then exposes properties and methods that allow other components to interact with it." -> "Interaction with the widget is possible from other components through the properties and methods that the widget exposes."

File: site/source/tutorials/002_creating_an_application/index.md [hunk 4]
```diff
@@ -28,58 +28,62 @@ You also need to be familiar with TypeScript as Dojo uses it extensively.
...
 
 The first diagram shows a traditional HTML + JavaScript architecture. Since the visual (HTML) and behavioral (JavaScript) aspects of the application are publicly accessible, the application's components can be manipulated directly which can lead to the HTML and JavaScript getting out of sync with each other. Extensive test suites are often needed to make sure that this does not happen.
 
 The second image shows how widgets ensure that components only interact according to their design intent. The widget encapsulates its visual and behavioral aspects. It then exposes properties and methods that allow other components to interact with it. By providing a controlled interface, it is much easier to keep the visual and behavioral aspects of the widget synchronized.
 
-In our demo application, we only have one widget, the HelloWorld widget:
+In our demo application, we don't currently have any widgets, but let's abstract the virtual DOM from `main.ts` out into a dedicated widget. You'll notice that to render a widget, we use a different pragma, `w()`, than we used before to render nodes that directly represent the DOM nodes.
```
Comment (dylans) [inline site/source/tutorials/002_creating_an_application/index.md]: we don't currently -> we do not currently

from `main.ts` out into -> from `main.ts` into

You'll -> You will

File: site/source/tutorials/002_creating_an_application/index.md [hunk 5]
```diff
@@ -28,58 +28,62 @@ You also need to be familiar with TypeScript as Dojo uses it extensively.
...
+This is very simple widget, containing the virtual DOM that we previously were returning straight from the vdom `renderer`, but it demonstrates some important concepts. Notice the `render` method, which provides the virtual nodes (also known as DNodes) for the `renderer` to determine what to add to the HTML document. In this example, the widget is simple enough that the function always returns the same result. We could make this widget more sophisticated by allowing it to take additional properties that can be used to alter how the DNodes are generated. We can provide sensible default values for these properties to add more complex behavior without having to change how the rest of the application interacts with it. This encourages the development of loosely coupled components that are easier to develop and maintain over time.
```
Comment (dylans) [inline site/source/tutorials/002_creating_an_application/index.md]: This is very simple widget -> This is a very simple widget

without having to change how the rest of the application interacts with it. -> without needing to change how the rest of the application interacts with the widget.

This encourages the development -> This approach encourages the development

File: site/source/tutorials/1030_routing/index.md [hunk 6]
```diff
@@ -38,21 +38,23 @@ All application routes needs to be to configured when creating the router instan
 * `defaultRoute` - a default route to be used if there is no other matching route
 * `children` - nested route configurations which can represent a nested path within a parent route
 
+The route configuration should be static, i.e. not dynamically determined at runtime and defined as the default export of a module called `routes.ts` in the projects `src` directory.
```
Comment (dylans) [inline site/source/tutorials/1030_routing/index.md]: projects `src` directory. -> project's `src` directory.

File: site/source/tutorials/1030_routing/index.md [hunk 7]
```diff
@@ -13,9 +13,9 @@ topic: routing
 
 ## Overview
 
-`@dojo/framework/routing` is a powerful set of tools to support declarative routing using a higher order component, an `Outlet` and a component that creates links with a `href` generated from an `outlet` name.
+`@dojo/framework/routing` is a powerful set of tools to support declarative routing using a specialized widget that accepts a render property called an `Outlet` and a widget that creates links with a `href` generated from an `outlet` id.
 
-In this tutorial, we will start with a basic application with no routing. We will use Dojo's declarative routing to configure some routes, create `outlets` from our existing widgets and use the `Link` component to create links for the application outlets.
+In this tutorial, we will start with a basic application with no routing. We will use Dojo's declarative routing to configure some routes, use `Outlet` widget to define the view for each route and use the `Link` widget to create links for the application's outlets.
```
Comment (dylans) [inline site/source/tutorials/1030_routing/index.md]: use `Outlet` widget -> use the `Outlet` widget

File: site/source/tutorials/1030_routing/index.md [hunk 8]
```diff
@@ -76,51 +78,62 @@ Next, we will create `outlets` to control when our widgets are displayed.
...
 
-{% task 'Create outlets for the routed widgets.' %}
+{% task 'Wrap the application widgets with the `Outlet` widget.' %}
 
 {% aside 'Reminder' %}
 The path that is associated to an outlet name is defined by the routing configuration from the first section of this tutorial.
 {% endaside %}
 
-An `Outlet` is a higher order component that wraps a widget, and controls whether the widget is rendered based on the navigated route associated to the outlets name. To keep our application organized, outlets are usually stored in a separate directory named `outlets`.
+The `Outlet` widget accepts two properties `id` and `renderer`, the `id` is the `outlet` from the routing configuration and the `renderer` is a function that returns widgets and nodes to display using `v()` and `w()`.
```
Comment (dylans) [inline site/source/tutorials/1030_routing/index.md]: two properties `id` -> two properties, `id` 

`renderer`, the `id` is the -> `renderer`. The `id` is the

File: site/source/tutorials/1030_routing/index.md [hunk 9]
```diff
@@ -76,51 +78,62 @@ Next, we will create `outlets` to control when our widgets are displayed.
...
 
-{% instruction 'Add the following code to `WorkerFormOutlet.ts`.' %}
+The `renderer` function receives a `MatchDetails` object that provides information about the route match.
 
-{% include_codefile 'demo/finished/biz-e-corp/src/outlets/WorkerFormOutlet.ts' %}
+ * router: The router instance, which can be used to generate links with, for example, `router.getLine('outlet-id')`.
```
Comment (dylans) [inline site/source/tutorials/1030_routing/index.md]: generate links with, for example, -> generate links, for example,

File: site/source/tutorials/1030_routing/index.md [hunk 10]
```diff
@@ -76,51 +78,62 @@ Next, we will create `outlets` to control when our widgets are displayed.
...
+Consider an `outlet` configured for a `path` of `about`, the widget that it returns from the `renderer` will render for a selected route `about` (described as an `index` match). The widget will also display for any route that the outlet's `path` partially matches, for example, `about/company` or `about/company/team`.
 
-Normally using a component as the first argument of `Outlet` is perfectly acceptable, however there are scenarios where it is necessary to explicitly define a widget for an `index` match. To support this more advanced configuration, the first argument also accepts an object that can be used to specify these components explicitly.
+Simply returning the widget or nodes that need to be displayed when an outlet has matched is usually all that is required, however there are scenarios where it is necessary to explicitly define a widget for an `index` or `error` match. This is where the `matchDetails` comes is, using the information from here we can write some simple logic to determine which widget to render for each scenario.
```
Comment (dylans) [inline site/source/tutorials/1030_routing/index.md]: This is where the `matchDetails` comes is, using the information from here we can write some simple logic to determine which widget to render for each scenario. -> This is where `matchDetails` is beneficial. By using the information from `matchDetails`, we can create simple logic to determine which widget to render for each scenario.

**Comment by agubler** [comment]
@dylans ready for round 2 of many, still need to update the link.

**Comment by agubler** [comment]
Updated link: https://dist-fovybkuhbd.now.sh/

**Pull request merged**